### PR TITLE
OpenSearch implementation

### PR DESCRIPTION
**Pull request by Newbytee**
## PR Checklist

Please check all that apply to this PR using "x":

- [X] I have checked that this PR is not a duplicate of an existing PR (open, closed or merged)
- [X] I have checked that this PR does not introduce a breaking change
- [ ] This PR introduces breaking changes and I have provided a detailed explanation below

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting)
- [ ] Refactoring (no functional changes)
- [ ] Documentation changes
- [ ] Other - Please describe:

## What is the current behavior?

lbry does not provide any OpenSearch search provider

## What is the new behavior?

lbry now provides an OpenSearch search provider

### CODE REVIEW

**Comment by tzarebczan** [comment]
This is awesome, thank you! Once you get this out of draft and we review, can we show you some [appreciation](https://lbry.com/faq/appreciation)?

Also please consider signing up for our dev mailing list at the bottom of lbry.tech!

File: static/opensearch.xml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<OpenSearchDescription xmlns="http://a9.com/-/spec/opensearch/1.1/">
+  <ShortName>lbry.tv</ShortName>
+  <Description>Search lbry.tv</Description>
+  <InputEncoding>UTF-8</InputEncoding>
+  <Image width="32" height="32" type="image/png">https://lbry.tv/public/favicon.png</Image>
```
Comment (kauffj) [inline static/opensearch.xml:5]: these need to be generated from `CONFIG` values, not hard coded

Comment (kauffj) [reply static/opensearch.xml:5]: true for all occurrences of lbry.tv

**Comment by Newbytee** [comment]
One thing I'd like to bring up is `insertVariableXml()` in `web/src/xml.js`. It is very similar to `insertToHead()` in `web/src/html.js`. I could make `insertToHead()` more generic and re-use it in both locations, but without doing that I think it would be confusing from a readability standpoint to use `insertToHead()` where `insertVariableXml()` is used at the moment.

**Comment by kauffj** [comment]
@seanyesmunt LGTM but I did not run this code

@Newbytee do you have a channel on LBRY? Checked for your handle on GitHub but did not see it

File: web/src/xml.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+const { URL, SITE_TITLE } = require('../../config.js');
+
+function getOpenSearchXml() {
+  return (
+    `<ShortName>${SITE_TITLE}</ShortName>` +
+    `<Description>Search ${SITE_TITLE}</Description>` +
+    '<InputEncoding>UTF-8</InputEncoding>' +
+    `<Image width="32" height="32" type="image/png">${URL}/public/favicon.png</Image>` +
```
Comment (kauffj) [inline web/src/xml.js:8]: I think this is also in the config?

Comment (Newbytee) [reply web/src/xml.js:8]: What are you referring to? Image size and type?

**Comment by Newbytee** [comment]
> @seanyesmunt LGTM but I did not run this code
> 
> @Newbytee do you have a channel on LBRY? Checked for your handle on GitHub but did not see it

I have an account but not a channel (seems I need some LBC to create a channel?).

File: web/src/xml.js (hunk 2 above)
Comment (Newbytee) [reply web/src/xml.js:8]: Looking at the source for the favicon it seems to have these properties hard-coded https://github.com/lbryio/lbry-desktop/blob/master/static/index-web.html#L7

**Comment by kauffj** [comment]
@Newbytee if you post an address here or email it to jeremy@lbry.com, I will send you some LBC. Thanks a lot for this contribution!

**Comment by Newbytee** [comment]
> @Newbytee if you post an address here or email it to [jeremy@lbry.com](mailto:jeremy@lbry.com), I will send you some LBC. Thanks a lot for this contribution!

Shot that address an email. Let me know if you want some sort of verification.

**Comment by seanyesmunt** [comment]
Looks good! @Newbytee do you have any idea how long it will take for this to be usable in browsers?

**Comment by seanyesmunt** [comment]
https://lbry.tv/opensearch.xml

**Comment by seanyesmunt** [comment]
That was quick. Just tested in chrome and it works!

**Pull request merged**